h5t_types.go

```diff
@@ -152,10 +153,14 @@ func (t *Datatype) Committed() bool {
 	return C.H5Tcommitted(t.id) > 0
 }
 
-// Copy copies an existing datatype. The returned datatype must be closed by the
-// user when it is no longer needed.
+// Copy copies an existing datatype.
 func (t *Datatype) Copy() (*Datatype, error) {
-	return copyDatatype(t.id)
+	c, err := copyDatatype(t.id)
```

Comment:
perhaps modify the signature of `copyDatatype` to take a `*Datatype` instead of `C.hid_t` so nobody would unadvertantly drop the `_hasGoPointer` value on the floor?

Reply:
Sure, I'll do that when I make `NewDatatype` do the same thing with `Identifier` (and check where there are others that may also benefit.